src/node/server/serverPluginClient.ts

```diff
@@ -1,15 +1,27 @@
 import fs from 'fs'
 import path from 'path'
 import chalk from 'chalk'
-import { ServerPlugin } from '.'
+import { Context, ServerPlugin, State } from '.'
 import { defaultDefines } from '../config'
+import Application from 'koa'
 
 export const clientFilePath = path.resolve(__dirname, '../../client/client.js')
 
 export const clientPublicPath = `/vite/client`
 
 const legacyPublicPath = '/vite/hmr'
 
+function updatePort(
+  clientCode: string,
+  ctx: Application.ParameterizedContext<State, Context>
+) {
+  // If the end-user is running the CLI within a GitHub Codespace,
+  // then set the port to 0, which indicates that we should use
+  // whichever port the client application is running on (e.g. 443).
+  const port = !!process.env['CODESPACES'] ? 0 : ctx.port
+  return clientCode.replace(`__PORT__`, port.toString())
```

Comment:
The `process.env['CODESPACES']` is not a commoned resolve case.Use the option configure it will be better for this.See https://github.com/vitejs/vite/pull/677/files.